### PR TITLE
Improve status monitor helpers

### PR DESCRIPTION
**Pull request by williamjamir**

### CODE REVIEW

File: alfasim_sdk/status.py [hunk 1]
```diff
@@ -1,16 +1,93 @@
...
+
+@attr.s(frozen=True)
+class PluginInfo:
+    name = attr.attrib(validator=non_empty_str)
+    enabled = attr.attrib(validator=instance_of(bool))
+    models = attr.attrib(validator=list_of_strings)
+
+
+@attr.s(frozen=True)
+class PipelineSegmentInfo:
```
Comment (ggrbill) [inline alfasim_sdk/status.py]: I am a little confused with this class. I mean, this class should have a parent information such as `Pipeline`, where we can identify where it ends. If there is no information about total length of the `Pipeline`, it is impossible to know where a `PipelineSegment` ends and what is its length.

File: alfasim_sdk/hook_specs_gui.py [hunk 2]
```diff
@@ -32,6 +32,47 @@ def alfasim_get_status(ctx) -> List[Union[WarningMessage, ErrorMessage]]:
...
     This checks can be used to guarantee the consistency of the data,
     or compatibility with some configuration made on ALFAsim.
+
+    Example:
+        from alfasim_sdk.status import ErrorMessage
+        @alfasim_sdk.hookimpl
+        def alfasim_get_status(ctx):
+            results = []
+
+            plugin_info_2 = [plugin for plugin in ctx.GetPluginsInfo() if plugin.name.endswith('Plugin2')][0]
```
[the remark this answers is not in the document]
Comment (williamjamir) [reply alfasim_sdk/hook_specs_gui.py]: Yep, I do agree =)

This docstring is outdated, I've created a field on the StudyContext called `GetPluginByName()`
I will update the docstring with the relevant info ;)

[the remark this answers is not in the document]
Comment (williamjamir) [reply alfasim_sdk/hook_specs_gui.py]: Commit b974189caba42e1395b29a135c9a04daa6a6d301 fix the docstring ;)

File: alfasim_sdk/status.py [hunk 3]
```diff
@@ -1,16 +1,93 @@
...
+    edge_name = attr.attrib(validator=non_empty_str)
+    inner_diameter = attr.attrib(validator=instance_of(Scalar))
+    start_position = attr.attrib(validator=instance_of(Scalar))
+
+
+@attr.s(frozen=True)
+class NodesInfo:
+    name = attr.attrib(validator=non_empty_str)
+    number_of_phases = attr.attrib(validator=optional(instance_of(int)))
```
[the remark this answers is not in the document]
Comment (williamjamir) [reply alfasim_sdk/status.py]: Commit 45c08cce5160c3493f661d275414d0f4059f4ab1 renames to `number_of_phases` to `number_of_phases_from_associated_pvt `

File: alfasim_sdk/status.py [hunk 4]
```diff
@@ -1,16 +1,93 @@
...
+
+
+@attr.s(frozen=True)
+class EdgesInfo:
+    name = attr.attrib(validator=non_empty_str)
+    number_of_phases = attr.attrib(validator=optional(instance_of(int)))
```
[the remark this answers is not in the document]
Comment (williamjamir) [reply alfasim_sdk/status.py]: Commit 45c08cce5160c3493f661d275414d0f4059f4ab1 renames to `number_of_phases` to `number_of_phases_from_associated_pvt `

File: alfasim_sdk/status.py [hunk 5]
```diff
@@ -1,16 +1,93 @@
...
+
+@attr.s(frozen=True)
+class PipelineSegmentInfo:
+    edge_name = attr.attrib(validator=non_empty_str)
+    inner_diameter = attr.attrib(validator=instance_of(Scalar))
+    start_position = attr.attrib(validator=instance_of(Scalar))
+
+
+@attr.s(frozen=True)
+class NodesInfo:
```
[the remark this answers is not in the document]
Comment (williamjamir) [reply alfasim_sdk/status.py]: Commit 649347b8a596df6d3d65c31d6ab1063c7ceb2192 rename `NodesInfo` to `NodeInfo` ;)

File: alfasim_sdk/status.py [hunk 6]
```diff
@@ -1,16 +1,93 @@
...
+    name = attr.attrib(validator=non_empty_str)
+    number_of_phases = attr.attrib(validator=optional(instance_of(int)))
+
+
+@attr.s(frozen=True)
+class EdgesInfo:
```
[the remark this answers is not in the document]
Comment (williamjamir) [reply alfasim_sdk/status.py]: Commit 649347b8a596df6d3d65c31d6ab1063c7ceb2192 rename `EdgesInfo` to `EdgeInfo` ;)

File: alfasim_sdk/status.py [hunk 7]
```diff
@@ -1,16 +1,93 @@
...
+class EmulsionModelType(Enum):
+    boxall2012 = "EmulsionModelType.boxall2012"
+    brauner2001 = "EmulsionModelType.brauner2001"
+    brinkman1952 = "EmulsionModelType.brinkman1952"
+    brinkman1952_and_yeh1964 = "EmulsionModelType.brinkman1952_and_yeh1964"
+    hinze1955 = "EmulsionModelType.hinze1955"
+    model_default = "EmulsionModelType.model_default"
+    mooney1951a = "EmulsionModelType.mooney1951a"
+    mooney1951b = "EmulsionModelType.mooney1951b"
+    no_model = "EmulsionModelType.no_model"
```
[the remark this answers is not in the document]
Comment (williamjamir) [reply alfasim_sdk/status.py]: Commit fd0564b38ee5354c3d9e79b8009df37bacd6895e sort the attributes as requested ;)

File: alfasim_sdk/status.py [hunk 8]
```diff
@@ -1,16 +1,93 @@
...
+    sleicher1962 = "EmulsionModelType.sleicher1962"
+    taylor1932 = "EmulsionModelType.taylor1932"
+
+
+class SolidsModelType(Enum):
+    mills1985_equilibrium = "SolidsModelType.mills1985_equilibrium"
+    no_model = "SolidsModelType.no_model"
```
[the remark this answers is not in the document]
Comment (williamjamir) [reply alfasim_sdk/status.py]: Commit fd0564b38ee5354c3d9e79b8009df37bacd6895e sort the attributes as requested ;)

File: alfasim_sdk/status.py (hunk 1 above)
Comment (williamjamir) [reply alfasim_sdk/status.py]: Commit be02ab45f6e7b045f5e1d9b127a97c9da9314ad9 adds a new field `PipelinInfo` thas has the `total_length` and a segments attribute ;)

File: alfasim_sdk/context.py [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+        edge_name: Name of the edge that this ``Pipeline`` is associate with.
+
+        segments: List of segments associates with this ``Pipeline``
+        For more information check `alfasim_sdk.context.PipelineSegmentInfo`
+
+        total_length: Total length of the pipeline.
+
+    """
+
+    name = attr.attrib(type=str, validator=non_empty_str)
```
Comment (oliveira-esss) [inline alfasim_sdk/context.py]: I'm not sure Pipeline names in ALFAsim will ever be different than "<None>".

Also:
Name associated* with...
``Pipeline`` is associated* with.

Comment (williamjamir) [reply alfasim_sdk/context.py]: Commit 87ba400 fix the grammar :)

Comment (oliveira-esss) [reply alfasim_sdk/context.py]: And what about the name property?

Comment (oliveira-esss) [reply alfasim_sdk/context.py]: Actually I just checked and they are named after the edge's name: "Edge 1 > Pipeline" or "My Custom Edge Name > Pipeline". But this is also an implementation detail, you can leave the _name_ in this class.

**Pull request merged**